sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastToCpuExec.scala

```diff
@@ -83,36 +83,36 @@ case class GpuBroadcastToCpuExec(override val mode: BroadcastMode, child: SparkP
             }
 
             // deserialize to host buffers in the driver and then convert to rows
-            val dataTypes = child.output.map(_.dataType)
-
-            val gpuBatches = serializedBatches.safeMap { cb =>
-              val hostColumns = (0 until cb.header.getNumColumns).map { i =>
-                val columnHeader = cb.header.getColumnHeader(i)
-                val hcv = new HostColumnVector(
-                  columnHeader.getType,
-                  columnHeader.rowCount,
-                  Optional.of(columnHeader.nullCount),
-                  cb.buffer,
-                  null, null, List.empty.asJava)
-                new RapidsHostColumnVector(dataTypes(i), hcv)
+            val gpuBatches = withResource(serializedBatches) { _ =>
+              serializedBatches.safeMap { cb =>
+                val hostColumns = (0 until cb.header.getNumColumns).map { i =>
+                  val columnHeader = cb.header.getColumnHeader(i)
+                  val hcv = new HostColumnVector(
+                    columnHeader.getType,
+                    columnHeader.rowCount,
+                    Optional.of(columnHeader.nullCount),
+                    cb.buffer,
+                    null, null, List.empty.asJava)
```

Comment:
I fixed the double-close issue, but this original code seems broken to me.  It's setting up all host columns in the batch to point to the _same data_ and no validity or offsets which seems very wrong.  this could never handle StringType which requires offsets or anything that could contain null values.